homeassistant/components/home_connect/api.py

```diff
@@ -348,31 +400,49 @@ class Oven(
     DeviceWithPrograms,
     DeviceWithRemoteControl,
     DeviceWithRemoteStart,
+    DeviceWithLocalControl,
+    DeviceWithProgramFinished,
 ):
     """Oven class."""
 
-    PROGRAMS = [
-        {"name": "Cooking.Oven.Program.HeatingMode.PreHeating"},
-        {"name": "Cooking.Oven.Program.HeatingMode.HotAir"},
-        {"name": "Cooking.Oven.Program.HeatingMode.TopBottomHeating"},
-        {"name": "Cooking.Oven.Program.HeatingMode.PizzaSetting"},
-        {"name": "Cooking.Oven.Program.Microwave.600Watt"},
-    ]
+    """Oven programs can't run by switches."""
 
     power_off_state = BSH_POWER_STANDBY
 
+    def get_oven_only_sensors(self):
```

Comment:
Could this be named `get_oven_temperature_sensor`  and references to it updated, so it is more descriptive? 
```suggestion
    def get_oven_temperature_sensors(self):
```